addons/web/static/src/core/l10n/numbers.js

```diff
@@ -86,11 +88,139 @@ export function parseFloat(value) {
+            throw new ParsingError(
+                sprintf(_lt("'%s' is not a correct float time").toString(), value)
+            );
+        } else {
+            throw e;
+        }
+    }
+    return floatTime;
+}
+
+/**
+ * Parse a more readable formated string representing an integer to an integer
+ *
+ * @param {string} value the integer to parse
+ * @returns integer
+ * @throws {Error} if the value can't be parsed properly
+ */
+export function parseInteger(value) {
+    let thousandsSepRegex = new RegExp(escapeRegExp(localization.thousandsSep), "g");
+    let decimalPointRegex = new RegExp(escapeRegExp(localization.decimalPoint), "g");
+    const parsed = parseNumber(value, {
+        thousandsSep: thousandsSepRegex,
+        decimalPoint: decimalPointRegex,
+    });
+    if (!Number.isInteger(parsed)) {
+        throw new ParsingError(sprintf(_lt("'%s' is not a correct integer").toString(), value));
     }
     return parsed;
 }
 
+/**
+ * Parse a string containing (optionally) a currency symbol and returns the amount as a float
+ *
+ * @param {string} value the amount to parse. It is assumed to be a pair {symbol, float} separated
+ *                by a non breaking space or a single float, e.g. $&nbsp;10 or 10&nbsp;€ or 10.
+ * @param {Object} [options={}]
+ * @param {number} [options.currencyId]
+ * @returns float
+ * @throws {Error} if no float is found or if the value does not have the format described above
+ */
+export function parseMonetary(value, options = {}) {
+    let amount;
+    try {
+        let values = value.split("&nbsp;");
+        if (values.length === 1) {
+            amount = parseFloat(value);
+        } else {
+            let currency;
+            if (options.currencyId) {
+                currency = odoo.session_info.currencies[options.currencyId];
+            } else {
+                currency = Object.values(odoo.session_info.currencies)[0];
+            }
+            const symbolIndex = values.findIndex((v) => v === currency.symbol);
+            if (symbolIndex === -1) {
+                throw new ParsingError();
+            }
+            values.splice(symbolIndex, 1);
+            if (values.length !== 1) {
+                throw new ParsingError();
+            }
+            amount = parseFloat(values[0]);
+        }
+    } catch (e) {
+        if (e instanceof ParsingError) {
+            throw new ParsingError(
+                sprintf(_lt("'%s' is not a correct monetary value").toString(), value)
+            );
+        } else {
+            throw e;
+        }
+    }
+    return amount;
+}
+
+/**
+ * Parse a string containing a float and unconvert it with a conversion factor of 100.
+ * The percentage can be a regular xx.xx float or a xx%.
+ *
+ * @param {string} value the percentage to parse
+ * @returns float
+ * @throws {Error} if no float is found or if the value does not have the format described above
+ */
+export function parsePercentage(value) {
+    let percentage;
+    try {
```

Comment:
why do you do a try catch? is it really worth it to customize the error message?